### PR TITLE
add methods for getting diagnostics report to health endpoint

### PR DESCRIPTION
**Pull request by margaret**
https://jira.mesosphere.com/browse/QUALITY-1819

Corresponding dcos PR https://github.com/dcos/dcos/pull/2485

### CODE REVIEW

**Comment by margaret** [comment]
Currently broken because it needs to be updated to use the `ARNodeApiClientMixin`.

File: dcos_test_utils/diagnostics.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    ARNodeApiClientMixin,
+    ApiClientSession,
+    RetryCommonHttpErrorsMixin,
+    check_json
+)
+
+log = logging.getLogger(__name__)
+
+
+class Diagnostics(ARNodeApiClientMixin, RetryCommonHttpErrorsMixin, ApiClientSession):
```
Comment (mellenburg) [inline dcos_test_utils/diagnostics.py]: Would it make sense to make this a subclass of DcosApiSession? Its fairly confusing that `all_slaves` is required just to support the ARNodeApiClientMixin, so it feels like it should be bound to an instantiation that supports that

File: dcos_test_utils/diagnostics.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    def wait_for_diagnostics_job(self, last_datapoint):
+        """
+        initial value of last_datapoint should be
+        last_datapoint = {
+            'time': None,
+            'value': 0
+        }
+        """
+        session_response = self.get('report/diagnostics/status/all')
+        print(session_response)
```
Comment (colin-msphere) [inline dcos_test_utils/diagnostics.py]: Is this debug?

File: dcos_test_utils/diagnostics.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        print(session_response.json())
```
Comment (colin-msphere) [inline dcos_test_utils/diagnostics.py]: Is this debug?

File: dcos_test_utils/diagnostics.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            assert 'is_running' in attributes, '`is_running` field is missing in response'
+            assert 'job_progress_percentage' in attributes, '`job_progress_percentage` field is missing'
+
+            if attributes['is_running']:
+                percent_done = attributes['job_progress_percentage']
+                logging.info("Job is running. Progress: {}".format(percent_done))
+                job_running = True
+                break
+
+        # if we ran this bit previously, compare the current databpoint with the one we saved
```
Comment (colin-msphere) [inline dcos_test_utils/diagnostics.py]: nit: `databpoint`

File: dcos_test_utils/diagnostics.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        def _at_least_one_item(bundle):
+            return bundle is not None and isinstance(bundle, list) and len(bundle) > 0
+
+        bundles = []
+        for bundle_list in response.values():
+            if _at_least_one_item(bundle_list):
+                bundles += map(lambda s: os.path.basename(s['file_name']), bundle_list)
+        return bundles
+
+    @retrying.retry(stop_max_delay=50000)
```
Comment (mellenburg) [inline dcos_test_utils/diagnostics.py]: Need to make this more qualified (i.e. don't retry every exception or else things like syntax errors get retried)

File: dcos_test_utils/helpers.py [hunk 6]
```diff
@@ -200,6 +212,34 @@ def retry_errors():
...
+        """ Communicating with a DC/OS cluster is done by default through Admin Router.
+        Use this Mixin with an ApiClientSession that requires distinguishing between nodes.
+        Admin Router has both a master and agent process and so this wrapper accepts a
+        node argument. node must be a host in self.master or self.all_slaves. If given,
+        the request will be made to the Admin Router endpoint for that node type
+        """
+        if node is not None:
+            assert port is None, 'node is intended to retrieve port; cannot set both simultaneously'
+            assert host is None, 'node is intended to retrieve host; cannot set both simultaneously'
+            if node in self.masters:
```
Comment (mellenburg) [inline dcos_test_utils/helpers.py]: This requires some special state which currently is only guaranteed by DcosApiSession... could be good to have a note for that

**Pull request merged**